server/src/main/java/com/saralein/server/handler/FileHandler.java

```diff
@@ -1,39 +1,100 @@
 package com.saralein.server.handler;
 
 import com.saralein.server.FileHelper;
+import com.saralein.server.filesystem.IO;
 import com.saralein.server.protocol.Methods;
 import com.saralein.server.request.Request;
+import com.saralein.server.response.ErrorResponse;
 import com.saralein.server.response.Response;
-
+import javax.xml.bind.DatatypeConverter;
 import java.io.IOException;
-import java.nio.file.Files;
 import java.nio.file.Path;
-import java.nio.file.Paths;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
 
 public class FileHandler implements Handler {
     private final FileHelper fileHelper;
+    private final IO fileIO;
 
-    public FileHandler(FileHelper fileHelper) {
+    public FileHandler(FileHelper fileHelper, IO fileIO) {
         this.fileHelper = fileHelper;
+        this.fileIO = fileIO;
     }
 
     @Override
-    public Response handle(Request request) throws IOException {
-        String requestMethod = request.getMethod();
-        byte[] body = requestMethod.equals(Methods.GET.name()) ? createBody(request) : new byte[]{};
+    public Response handle(Request request) throws IOException, NoSuchAlgorithmException {
+        String method = request.getMethod();
+
+        if (isHeadRequest(method)) {
+            return getHeadResponse(request);
+        }
 
+        if (isPatchRequest(method)) {
+            return getPatchResponse(request);
+        }
+
+        return getGetResponse(request);
+    }
+
```

Comment:
Something about all these conditionals for different types makes me think this file handler is doing too much. Maybe we could utilize middleware and polymorphism? A couple types of FileHandlers?